File: app/src/main/java/com/stathis/elmepaunivapp/database/AnnouncementsDao.java
package com.stathis.elmepaunivapp.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.stathis.elmepaunivapp.ui.announcements.model.Announcement;

import java.util.List;

@Dao
public interface AnnouncementsDao {

    @Query("SELECT * FROM Announcements")
    public List<Announcement> getAll();

    @Insert
    void insert(Announcement announcement);

    @Update
    void update(Announcement announcement);

    @Delete
    void delete(Announcement announcement);

}

File: app/src/main/java/com/stathis/elmepaunivapp/database/AnnouncementsDao.kt
package com.stathis.elmepaunivapp.database

import androidx.room.*
import com.stathis.elmepaunivapp.model.Announcement
import java.util.*

@Dao
interface AnnouncementsDao {

    @Query("SELECT * FROM Announcements")
    fun getAll(): List<Announcement>

    @Query("DELETE from Announcements")
    fun deleteAll()

    @Insert
    fun insert(announcement: Announcement)

    @Insert
    fun insertAll(announcement: ArrayList<Announcement>)

    @Delete
    fun update(announcement: ArrayList<Announcement>)

    @Delete
    fun delete(announcement: Announcement)
}